### ISSUE TITLE
Mixed CPU/GPU Fabric instances 

### ISSUE THREAD

**Issue by patchmeifyoucan** [Lightning-AI/pytorch-lightning#18158]
### 📚 Documentation

Hello,

I am using Fabric on a SLURM cluster which has 4 GPUs and 2 Intel Xeon Platinum 8368 (total of 76 CPU cores) per node. 
My main use of Fabric so far was to discover devices and to create processes accordingly, as well as for DDP-like training. 

I now need to extend my application to use more manually manged distributed components which use Torch's RPC framework for communication. This will likely remove my usage of Fabric setting up models, optimizers etc. since I will probably implement a parameter server and send gradients around myself. But I will would like to use it for assigning ranks and for logging, callbacks etc. in the future. 

I am in a distributed RL context. I would like the GPU ranks to do all the heavy processing (all the forward passes, since they work on images), which is pretty slow on the CPUs. However, I would like the CPUs to do management work in the background such as being a parameter server or a distributed replay buffer.

My question is whether it is possible to "join" (or whatever is appropriate) multiple Fabric instances. Even though I could do some manual work here, it would be nice if it is somehow possible to get the proper ranks from Fabric rather than spawning new processes from the GPU ranks (which would work but also I don't really want them to be tied together). 

If there are technical reasons why it is not possible or too cumbersome to do such training that's also fine for me and something that I can take into account when changing my application. I read the docs but did not find anything related (since it's quite a special thing to do, I guess). 

It would be nice to hear your opinion on this. Thank you very much (and I hope the docs section is appropriate for this kind of requests). 

cc @borda
